modules/stochastic_tools/src/multiapps/SamplerFullSolveMultiApp.C

```diff
@@ -29,13 +30,118 @@ validParams<SamplerFullSolveMultiApp>()
   params.suppressParameter<std::vector<Point>>("move_positions");
   params.suppressParameter<std::vector<unsigned int>>("move_apps");
   params.set<bool>("use_positions") = false;
+
+  MooseEnum modes("normal=0 batch-reset=1 batch-restore=2", "normal");
+  params.addParam<MooseEnum>(
+      "mode",
+      modes,
+      "The operation mode, 'normal' creates one sub-application for each row in the Sampler and "
+      "'batch' creates on sub-application for each processor and re-executes for each row.");
+
   return params;
 }
 
 SamplerFullSolveMultiApp::SamplerFullSolveMultiApp(const InputParameters & parameters)
   : FullSolveMultiApp(parameters),
     SamplerInterface(this),
-    _sampler(SamplerInterface::getSampler("sampler"))
+    _sampler(SamplerInterface::getSampler("sampler")),
+    _mode(getParam<MooseEnum>("mode"))
+{
+  if (_mode == "batch-reset" || _mode == "batch-restore")
+    init(n_processors());
```

Comment:
Can each multiapp still be ran in parallel?

Reply:
Are you asking if the MultiApp, when running in batch mode, can use more that one processor? If so, then no, because of this line. Is this something you need. We could perhaps make a input parameter: "processors_per_multiapp".

Reply:
I think we will leave this for now, we can add this feature when needed.